src/index.js

```diff
@@ -3,103 +3,105 @@ const http = require('http')
 const express = require('express')
 const socketio = require('socket.io')
 const Filter = require('bad-words')
-const { generateMessage, generateLocationMessage } = require('./utils/messages')
+
+const {
+    generateWelcome,
+    generateJoined,
+    generateLeft,
+    makeMessageGenerator,
+} = require('./utils/messages')
+
 const {
-    addUser,
-    removeUser,
-    getUser,
-    getUsersInRoom,
-    getRooms,
-} = require('./utils/users')
+    userToRoomJoiner,
+    roomDataFetcher,
+    userFromRoomDropper,
+} = require('./services')
+const { roomRepo } = require('./repos')
+const { Location } = require('./entities')
 
 const app = express()
 const server = http.createServer(app)
 const io = socketio(server)
 
-const port = process.env.PORT
-
 const publicDirPath = path.join(__dirname, '../public')
+const filter = new Filter()
```

Comment:
Do I understand correctly that you removed this from `socket.on('sendMessage')` because that way it swamps the memory (being created on each message)?

Reply:
Exactly !